daemon/src/order/maker.rs

```diff
@@ -0,0 +1,383 @@
+        // -> Should be fine because the offer is already outdated by then and we should actually
+        // wait until finished?
+        // -> Cannot use send_async_next here - is that expected?
+        let latest_offers = match self
+            .latest_offers
+            .send(xtra_libp2p_offer::maker::GetLatestOffers)
+            .await
+        {
+            Ok(latest_offers) => latest_offers,
+            Err(e) => {
+                bail!("Failed to retrieve latest offer from offers actor: {e:#}")
+            }
+        };
+
+        let offer = match latest_offers {
+            None => bail!("No latest offers available"),
+            Some(offers) => offers.pick_offer_to_take(offer_id),
+        };
+
+        match offer {
+            None => bail!("Offer with id {offer_id} not found in current offers"),
+            Some(offer) => Ok(offer),
+        }
+    }
+}
+
+#[xtra_productivity(message_impl = false)]
+impl Actor {
+    async fn handle(&mut self, msg: NewInboundSubstream, ctx: &mut xtra::Context<Self>) {
+        let NewInboundSubstream { peer, stream } = msg;
+
+        let mut framed = Framed::new(stream, JsonCodec::<MakerMessage, TakerMessage>::new());
+
+        let order = match self.await_order(&mut framed).await {
+            Ok(order) => order,
+            Err(e) => {
+                tracing::error!("Failed to receive order from taker: {e:#}");
+                return;
+            }
+        };
+
+        let (id, offer, quantity, leverage, taker_id) = match order {
+            TakerMessage::PlaceOrder {
+                id,
+                offer: offer_id,
+                quantity,
+                leverage,
+                taker_id,
+            } => (id, offer_id, quantity, leverage, taker_id),
+            TakerMessage::ContractSetupMsg(_) => {
+                tracing::error!("Unexpected message");
+                return;
+            }
+        };
+
+        tracing::info!(taker = %peer, %quantity, order_id = %id, "Taker wants to place an order");
+
+        // Reject the order if the offer cannot be found in the latest offers
+        let offer = match self.pick_offer(offer.id).await {
+            Ok(offer) => offer,
+            Err(e) => {
+                tracing::warn!("Rejecting taker order because unable to pick offer: {e:#}");
+
+                let future = async move {
+                    framed
+                        .send(MakerMessage::Decision(protocol::Decision::Reject))
+                        .await?;
+
+                    anyhow::Ok(())
+                };
+
+                tokio_extras::spawn_fallible(
+                    &ctx.address().expect("self to be alive"),
+                    future,
+                    move |e| async move {
+                        tracing::debug!(%peer, "Failed to send reject order message: {e}");
+                    },
+                );
+
+                return;
+            }
+        };
```

Comment:
I agree. This was discussed to some extent, but it would be a change that requires adapting the automation. I would like to keep this changeset contained and just keep the logic as it was before, and then change this in another iteration.

This is a change that purely concerns the maker, so we can change this easily on top, I will see to create a ticket.

Reply:
Note: I am happy to action this straight away, so we also solve the problem of not needing to know the latest offer in this actor. 
But, we have to evaluate how the automation then keeps the state of the current offer. Currently it just does not keep that state as far as I know, so I don't think this is such a simple change and would not see it in scope of this PR.

Reply:
Thanks for your reply! I think it's fine to not tackle it immediately, particularly if it's not a simple change. If you create a ticket and we tackle it soon I will be more than happy.

Reply:
Created :) see linked tickets